working/extension_structs/overview.md

```diff
@@ -0,0 +1,815 @@
+# Overview of a proposal for structs
+
+Author: Leaf Petersen
+
+Status: In progress
+
+Version 1.0 (see [CHANGELOG](#CHANGELOG) at end)
+
+## Summary
+
+This document is an overview of an approach to solving the following problems:
+  - The desire for a zero cost wrapper type (motivated largely by interop
+   concerns), described briefly
+   [here](https://github.com/dart-lang/language/issues/1474)
+  - The desire to have compact syntax for so called "data classes" or "value
+types", discussed among other places
+[here](https://github.com/dart-lang/language/issues/314)
+  - The desire to have an accounting for data structures without identity as
+    described [here](https://github.com/dart-lang/language/issues/2246).
+
+The proposal here takes two steps.  The first is to add a restricted kind of
+class (provisionally describe as a "struct" here), which gives up some of the
+affordances of general Dart classes in exchange for compact syntax and automatic
+generation of useful methods.  These restricted classes provide data class like
+functionality.  The second step is to support a further restriction on structs
+with a single field which eliminates the wrapper object, representing the struct
+entirely as the underlying object (at the cost of making the abstraction
+entirely static).
+
+This proposal builds on previous proposals in this space, including:
+  - [Views](https://github.com/dart-lang/language/blob/master/working/1426-extension-types/feature-specification-views.md)
+  - [Extension types](https://github.com/dart-lang/language/blob/master/working/1426-extension-types/feature-specification.md)
+  - [Protected extension types](https://github.com/dart-lang/language/issues/1467)
+
+
+This overview is not intended to be a feature specification: it is designed to
+give a brief overview of the core proposal and set out some design points for
+discussion.  If there is buy-in on these ideas, we may choose to incorporate
+them into an existing proposal, or add a new feature specification.
+
+
+## Design principles
+
+We aim to minimize the differences between structs and classes.  As much as
+possible, structs should behave as restrictions of classes, and extension
+structs should behave as further restrictions of structs.  We specifically aim
+to avoid as much as possible having different behaviors for the same concept
+(e.g. differences in scoping).
+
+We also aim to minimize the amount of new syntax required, and to maximize the
```

Comment:
Sorry I missed this sentence on my earlier pass. I think this aligns with my other comment about the primary constructor syntax and whether we should consider that a second-tier syntax.